### ISSUE TITLE
v-el and v-ref bindings should be reactive

### ISSUE THREAD

**Issue by cheapsteak** [yyx990803/vue#1102]
When a `<component>` with a `v-ref` changes its content to another component, and the two components have the same property with different values, manually accessing the property via $.ref.property returns the correct value, but data-bindings are not updated

http://jsfiddle.net/t2u6bvqh/5/

```html
<component is="{{componentId}}" v-ref="child"></component>
{{$.child.val}} <-- this value will never be updated (essentially a one-time binding)
```

**Comment by yyx990803** [comment]
Both this and #1011 actually share the same root cause: `v-ref` and `v-el` bindings are not reactive. Renaming this one and closing #1011.

**Comment by yyx990803** [comment]
Implemented in 1.0.0-alpha.1.
